render/wlr_renderer.c

```diff
@@ -6,6 +7,15 @@
 
 void wlr_renderer_init(struct wlr_renderer *renderer,
 		const struct wlr_renderer_impl *impl) {
+	assert(impl->begin);
+	assert(impl->clear);
+	assert(impl->scissor);
+	assert(impl->render_texture_with_matrix);
+	assert(impl->render_quad);
+	assert(impl->render_ellipse);
+	assert(impl->formats);
+	assert(impl->format_supported);
+	assert(impl->texture_from_pixels);
```

Comment:
:+1:, but let's not make a habit of this - some types are allowed to implement a subset of the interface.

Reply:
Actually, I made some of them optional - mainly the `wl_drm` stuff.